src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -4909,7 +4901,7 @@ lElseClause:
             Return New BoundAwaitOperator(node, operand,
                                           awaitableInstancePlaceholder, getAwaiter,
                                           awaiterInstancePlaceholder, isCompleted, getResult,
-                                          resultType, hasErrors)
+                                          type:=getResult.Type, hasErrors)
```

Comment:
>getResult.Type [](http://example.com/codeflow?start=48&length=14)

It will be good to have a test for a scenario when we are awaiting a ```Task``` rather than a ```Task(Of T)```.

Reply:
Also it would be good to test the effect of this change on an `await` in a Sub expression lambda.

Reply:
> a scenario when we are awaiting a Task rather than a Task(Of T).

I think `TestAwaitExpression` (first test in that file) covers that.

> an await in a Sub expression lambda.

Can you clarify what you'd like this test to verify? Just the IOperation node for the await in statement in lambda, or something else?

Reply:
> Can you clarify what you'd like this test to verify? Just the IOperation node for the await in statement in lambda, or something else?

Emit, execution, optionally IOperation.